From 5d5bda27f4ba9428b65a519cfe1342720ff41324 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:08:21 -0400
Subject: [usb] sierra: fix line-control pipe direction

Message-id: <1410887463-20674-430-git-send-email-dzickus@redhat.com>
Patchwork-id: 95260
O-Subject: [RHEL7 PATCH 429/591] USB: sierra: fix line-control pipe direction
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit d304889888af2ef65c94dfb52441cfecb05853e8
Author: Johan Hovold <jhovold@gmail.com>
Date:   Mon May 26 19:22:58 2014 +0200

    USB: sierra: fix line-control pipe direction

    The sierra line-control request has been using the wrong pipe direction,
    while relying on USB core to fix it up.

    Signed-off-by: Johan Hovold <jhovold@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/serial/sierra.c b/drivers/usb/serial/sierra.c
index 0254f6d..4cb11b7 100644
--- a/drivers/usb/serial/sierra.c
+++ b/drivers/usb/serial/sierra.c
@@ -365,7 +365,7 @@ static int sierra_send_setup(struct usb_serial_port *port)
  if (retval < 0)
   return retval;
 
- retval = usb_control_msg(serial->dev, usb_rcvctrlpipe(serial->dev, 0),
+ retval = usb_control_msg(serial->dev, usb_sndctrlpipe(serial->dev, 0),
   0x22, 0x21, val, interface, NULL, 0, USB_CTRL_SET_TIMEOUT);
  usb_autopm_put_interface(serial->interface);
 
-- 
1.7.1